Sources/COWMacros/COWExpansionFactory.swift

```diff
@@ -218,30 +223,196 @@ internal class COWExpansionFactory<Context: MacroExpansionContext> {
   
   // MARK: Utilities
   
-  private func createDerivedStorageTypeDecl() -> StructDeclSyntax {
-    let members = appliedStructValidVarDecls.map {
-      MemberBlockItemSyntax(decl: $0)
+  // If `appliedStructDecl` conforms to Equatable and manually implements `==`,
+  // the derived storage must also conform to Equatable (otherwise we hit a
+  // compiler bug that breaks the build).
+  // https://forums.swift.org/t/macro-multiple-matching-functions-xyz-error-though-the-expanded-code-has-no-duplicate-functions-and-compiles/68700/4
+  // FIXME: remove the workaround when the compiler bug is fixed.
+  private func applyEquatableWorkaroundForDerivedStorage(
+    members: inout [MemberBlockItemSyntax],
+    protocols: inout [InheritedTypeSyntax]
+  ) {
+    // Bail out if derived storage doesn't conform to Equatable.
+    guard protocols.contains(where: {
+      guard let name = $0.type.identifier else {
+        return false
+      }
+      return equatableProtocolNames.contains(name) ||
+             hashableProtocolNames.contains(name)
+    }) else {
+      return
     }
     
+    // Bail out if `appliedStructDecl` does not manually implement `==`.
+    guard let nestingEqualFunc = appliedStructDecl.memberBlock.members
+      .lazy
+      .map(\.decl)
+      .compactMap({$0.as(FunctionDeclSyntax.self)})
+      .first(where: {$0.likelyToConformToEquatable(for: appliedStructDecl)}) else {
+      return
+    }
+    
+    // Another compiler bug another workaround.
+    // The compiler doesn't call the manually-implemented `==` in `appliedStructDecl`;
+    // it will synthesize one for the derived storage and call it instead.
+    // Expand an `==` that is an exact copy (except for the signature) of
+    // `appliedStructDecl` to make sure the program behavior is correct.
+    
+    // In order to make this workaround work with computed properties and
+    // immutable functions with return values, we also have to copy them into
+    // the derived storage.
+    appliedStructDecl.memberBlock.members
+      .lazy
+      .map(\.decl)
+      .compactMap { $0.as(VariableDeclSyntax.self) }
+      .filter { !$0.isStatic && !$0.isStored }
+      .forEach { members.append(MemberBlockItemSyntax(decl: $0)) }
+    appliedStructDecl.memberBlock.members
+      .lazy
```

Comment:
What's the purpose of this `.lazy`? The result is evaluated in place by calling to `forEach` at the end of this call chain.